00_Base/src/ocpp/model/enums/index.ts

```diff
@@ -808,7 +808,7 @@ export const enum UpdateEnumType {
  * within the CSMS.
  *
  */
-export const enum RegistrationStatusEnumType {
+export enum RegistrationStatusEnumType {
```

Comment:
Out of curiosity, what triggered this change. From why I understand, the const does an inline replacement. But I am not quite aware of the other tradeoffs here.

Reply:
const enums are not available at run time. Essentially, those enums get transpiled as explicit values and the enum itself ends up not being available at run time. So when you do code like:
```
if (someVal === SeomEnum.SomeVal) {
  ...
}
```
for example, in your code, it will work, but the JS code will be something along the lines of
```
if (someVal === 'SomeVal') {
  ...
}
```
which can be fine.. but I was facing an issue where if this particular enum is used in an annotation, for example like so: 
```
@IsNotEmpty()
@IsEnum(RegistrationStatusEnumType)
@Enum(RegistrationStatusEnumType, 'RegistrationStatusEnumType')
unknownChargerStatus: RegistrationStatusEnumType;
```
then it does not accept a `const enum`...

Reply:
Ah, got it. For you it was around the annotations. I ran into this when creating tests and fakers. 
So probably the inlining (your second code snippet) is more efficient. But maybe not worth the efficiency gain for a node backend application against keeping it as an object where you can to more operations against.

Reply:
I think in cases where the enum is quite large and used in a lot of places, perhaps the performance gain, of not having the enum object in run-time but instead having static strings everywhere in the running JS can have some impact, especially if it is compounded with other enums as the project scales and in partifular for enums that have high frequency of lookups since the `const enums` will be returning values directly while `enums` need to look it up in the object. On the other hand, if you have a small enum that is only used in a small hand full of places, rarely running, and you need to access the actual enum mapping, for example, if you ever need to iterate over the enum, or if you want to have a dynamic `valueFrom(str)` then with the `const enum` approach that can only be achieved via a switch or if statements. I guess what I am trying to say is that there is most likely a balance that needs to be struck identifying enums that need to exist during run-time and their use-cases, vs the enums that should just be transpiled into const strings.